### DISCUSSION TITLE
Bodies-of-pure-lists throws 400

### DISCUSSION THREAD

**Discussion by ramki88** [fastapi/fastapi#10461] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel, HttpUrl

app = FastAPI()


class Image(BaseModel):
    url: HttpUrl
    name: str


@app.post("/images/multiple/")
async def create_multiple_images(images: list[Image]):
    return images
```

### Description

Request:
POST /images/multiple/ HTTP/1.1
Host: localhost:8000
Content-Type: application/json
Accept: application/json
[
    {
        "url": "https://localhost",
        "name": "Pic"
    },
    {
        "url": "https://localhost",
        "name": "Pic2"
    }
]

Response:
{
    "detail": "There was an error parsing the body"
}

### Operating System

macOS

### FastAPI Version

0.103.2 

### Pydantic Version

1.10.13

### Python Version

3.11.6

**Comment by hasansezertasan** [comment, marked as answer]
It works great on my system:

Python 3.11.4
Pydantic 2.4.2
FastAPI 0.103.2

Try to upgrade your pydantic version, use an environment if you aren't using one, preferably virtualenv.

**Reply by ramki88** [reply]
@hasansezertasan, Thanks for quickly checking. My bad, I had a route_class which was causing the issue 😞

**Reply by hasansezertasan** [reply]
Happy to help 🙏
